File: Javascript/Divide and Conquer/Min-max Finding/iterative-method.js
// Define the function for finding the minimum and maximum elements in an array
function minMax(myList, len) {
    let min = myList[0];
    let max = myList[0];

    for (let i = 1; i < len; i++) {
        if (myList[i] > max) {
            max = myList[i];
        }
        if (myList[i] < min) {
            min = myList[i];
        }
    }

    return [min, max];
}

// Define the main function
function main() {
    let myList = [5, 8, 7, 4, 2, 9, 12, 13, 11, 23, 1, -1, -23, 34, 545];
    let n = myList.length;

    let [min, max] = minMax(myList, n);
    console.log(`min = ${min}, max = ${max}`);
}

// Call the main function
main();

File: Python/Divide and Conquer/Min-Max Finding/iterative-method.py
def min_max(my_list, len):
    min = my_list[0]
    max = my_list[0]

    for i in range(1, len):
        if my_list[i] > max:
            max = my_list[i]
        if my_list[i] < min:
            min = my_list[i]
    

    return min, max

def main():
    my_list = [5, 8, 7, 4, 2, 9, 12, 13, 11, 23, 1, -1, -23, 34, 545]
    n = len(my_list)

    min, max = min_max(my_list, n)
    print(f"min = {min}, max = {max}")


if __name__ == "__main__":
    main()
